### PR TITLE
Parallel upload of object parts

### LINKED ISSUE #503: Parallel upload through fput_object API
Uploading a 7.5 gig file through mc client takes 30 seconds, while minio-py API fput_object takes about 3 minutes for the same task.
There should be an option for parallel upload in the API as well which can match the mc client performance.

### PR DESCRIPTION
**Pull request by vadmeste**
Fixes #503 

Based on the first attempt here https://github.com/minio/minio-py/pull/536

### CODE REVIEW

File: tests/functional/tests.py [hunk 1]
```diff
@@ -15,9 +15,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import sys
-sys.path.append('/home/vadmeste/work/python/minio-py')
```
Comment (harshavardhana) [inline tests/functional/tests.py]: oh wow when was this added?

Comment (vadmeste) [inline tests/functional/tests.py]: > oh wow when was this added?

Few weeks ago, it was when I restructured tests, which looks like a big change in diff.

Comment (harshavardhana) [inline tests/functional/tests.py]: Can you send fix for this quickly? lets get it merged before merging this PR?

**Comment by vadmeste** [comment]
@donatello, can you review this ?

**Pull request merged**